GDJS/Runtime/ResourceLoader.ts

```diff
@@ -0,0 +1,566 @@
+        );
+        return;
+      }
+      let loadedCount = 0;
+      await Promise.all(
+        [...sceneResources.values()].map(async (resourceName) => {
+          const resource = this._resources.get(resourceName);
+          if (!resource) {
+            logger.warn('Unable to find resource "' + resourceName + '".');
+            return;
+          }
+          await this._loadResource(resource);
+          loadedCount++;
+          this.currentSceneLoadingProgress = loadedCount / this._resources.size;
+          onProgress && (await onProgress(loadedCount, this._resources.size));
+        })
+      );
+      this._setSceneAssetsLoaded(sceneName);
+      console.log('Done: ' + sceneName);
+    }
+
+    private async _loadResource(resource: ResourceData): Promise<void> {
+      const resourceManager = this._resourceManagersMap.get(resource.kind);
+      if (!resourceManager) {
+        logger.warn(
+          'Unknown resource kind: "' +
+            resource.kind +
+            '" for: "' +
+            resource.name +
+            '".'
+        );
+        return;
+      }
+      console.log('Load: ' + resource.name);
+      await resourceManager.loadResource(resource.name);
+    }
+
+    /**
+     * Load and process a scene that is needed right away.
+     *
+     * The renderer will show a loading screen while its done.
+     */
+    async loadAndProcessSceneResources(
+      sceneName: string,
+      onProgress?: (count: number, total: number) => Promise<void>
+    ): Promise<void> {
+      if (this.areSceneAssetsReady(sceneName)) {
+        return;
+      }
+      await this.loadSceneResources(sceneName, onProgress);
+
+      const sceneResources = this._sceneResources.get(sceneName);
+      if (!sceneResources) {
+        logger.warn(
+          'Can\'t load resource for unknown scene: "' + sceneName + '".'
+        );
+        return;
+      }
+
+      let parsedCount = 0;
+      for (const resourceName of sceneResources) {
+        const resource = this._resources.get(resourceName);
+        if (!resource) {
+          logger.warn('Unable to find resource "' + resourceName + '".');
+          continue;
+        }
+        await this._processResource(resource);
+        parsedCount++;
+        onProgress && (await onProgress(parsedCount, sceneResources.length));
+      }
+      this._setSceneAssetsReady(sceneName);
+    }
+
+    /**
+     * Load a scene resources without parsing them.
+     *
+     * When another scene resources are loading in background, it waits for
+     * all its resources to be loaded before loading resources of the given
+     * scene.
+     */
+    async loadSceneResources(
+      sceneName: string,
+      onProgress?: (count: number, total: number) => void
+    ): Promise<void> {
+      const task = this._prioritizeScene(sceneName);
+      return new Promise<void>((resolve, reject) => {
+        if (!task) {
+          console.log('Already downloaded scene: ' + sceneName);
```

Comment:
Let's remove all the console.log - or if you want to keep some (which is fine), use `logger`